### ISSUE TITLE
Error deploying app with Apache CXF

### ISSUE THREAD

**Issue by johny65** [payara/Payara#4710]
# Description #
----------

I can't deploy an app providing a web service with Apache CXF because it fails with an exception.

## Expected Outcome

Being able to deploy the app and use the Apache CXF implementation to provide the web services.

## Current Outcome

I've read that in order to make use of the CXF implementation, I must configure this in the `glassfish-web.xml`file:

    <class-loader delegate="false"/>

But with that line, the app doesn't deploy throwing the following exception:

```
[2020-06-11T14:00:32.025-0300] [Payara 5.201] [GRAVE] [NCLS-CORE-00026] [javax.enterprise.system.core] [tid: _ThreadID=119 _ThreadName=admin-thread-pool::admin-listener(1)] [timeMillis: 1591894832025] [levelValue: 1000] [[
  Exception during lifecycle processing
java.lang.Exception: java.lang.IllegalStateException: ContainerBase.addChild: start: org.apache.catalina.LifecycleException: org.apache.catalina.LifecycleException: javax.servlet.ServletException: com.sun.xml.ws.transport.http.servlet.WSServletException: WSSERVLET11: Fallo al analizar el descriptor en tiempo de ejecución: com.sun.xml.ws.util.ServiceConfigurationError: com.sun.xml.ws.policy.jaxws.spi.PolicyFeatureConfigurator: Provider com.sun.xml.ws.transport.tcp.policy.TCPTransportFeatureConfigurator is specified in bundle://a41f04f5-90ca-4352-80f3-a3b08d61f75a_391.0:0/META-INF/services/com.sun.xml.ws.policy.jaxws.spi.PolicyFeatureConfiguratorbut could not be instantiated: java.lang.ClassCastException: Cannot cast com.sun.xml.ws.transport.tcp.policy.TCPTransportFeatureConfigurator to com.sun.xml.ws.policy.jaxws.spi.PolicyFeatureConfigurator
	at com.sun.enterprise.web.WebApplication.start(WebApplication.java:137)
	at org.glassfish.internal.data.EngineRef.start(EngineRef.java:123)
	at org.glassfish.internal.data.ModuleInfo.start(ModuleInfo.java:283)
	at org.glassfish.internal.data.ApplicationInfo.start(ApplicationInfo.java:362)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.initialize(ApplicationLifecycle.java:621)
	at org.glassfish.deployment.admin.DeployCommand.execute(DeployCommand.java:565)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:558)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:554)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/javax.security.auth.Subject.doAs(Subject.java:361)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2.execute(CommandRunnerImpl.java:553)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:584)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:576)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/javax.security.auth.Subject.doAs(Subject.java:361)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:575)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:1496)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.access$1300(CommandRunnerImpl.java:120)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1878)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1754)
	at org.glassfish.admin.rest.resources.admin.CommandResource.executeCommand(CommandResource.java:409)
	at org.glassfish.admin.rest.resources.admin.CommandResource.execCommandSimpInMultOut(CommandResource.java:236)
	at jdk.internal.reflect.GeneratedMethodAccessor672.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:124)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:167)
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:176)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:79)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:469)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:391)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:80)
	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:253)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:232)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:680)
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:356)
	at org.glassfish.admin.rest.adapter.JerseyContainerCommandService$3.service(JerseyContainerCommandService.java:179)
	at org.glassfish.admin.rest.adapter.RestAdapter.service(RestAdapter.java:189)
	at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:520)
	at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:217)
	at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:182)
	at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:156)
	at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:218)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:95)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:260)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:177)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:109)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:88)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:53)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:524)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:89)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:94)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:33)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:114)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:569)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:549)
	at java.base/java.lang.Thread.run(Thread.java:834)
]]
```

I don't know if this is a bug of Payara, Apache CXF, a problem with configuration or what.

## Steps to reproduce (Only for bug reports) 

I provide a simple project to reproduce [(test-cxf.zip)](https://github.com/payara/Payara/files/4766250/test-cxf.zip). Just:

    $ gradle clean build

    $ asadmin deploy --contextroot=/testws build/libs/test-cxf-0.1.war

## Environment ##

- **Payara Version**: 5.201
- **Edition**: Full
- **JDK Version**: openjdk version "11.0.7"
- **Operating System**: Linux (Ubuntu 20.04)

**Comment by johny65** [comment]
If I try to deploy the example in Payara Micro, I get:

```
Exception during lifecycle processing
java.lang.Exception: java.lang.IllegalStateException: ContainerBase.addChild: start: org.apache.catalina.LifecycleException: org.apache.catalina.LifecycleException: javax.servlet.ServletException: com.sun.xml.ws.transport.http.servlet.WSServletException: WSSERVLET11: Fallo al analizar el descriptor en tiempo de ejecución: java.lang.NoClassDefFoundError: com/sun/org/apache/xml/internal/resolver/CatalogManager
	at com.sun.enterprise.web.WebApplication.start(WebApplication.java:137)
	at org.glassfish.internal.data.EngineRef.start(EngineRef.java:123)
	at org.glassfish.internal.data.ModuleInfo.start(ModuleInfo.java:283)
	at org.glassfish.internal.data.ApplicationInfo.start(ApplicationInfo.java:362)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.initialize(ApplicationLifecycle.java:621)
	at fish.payara.deployment.admin.InitializeAllApplicationsCommand.execute(InitializeAllApplicationsCommand.java:75)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:558)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:554)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/javax.security.auth.Subject.doAs(Subject.java:361)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2.execute(CommandRunnerImpl.java:553)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:584)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:576)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/javax.security.auth.Subject.doAs(Subject.java:361)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:575)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:1496)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.access$1300(CommandRunnerImpl.java:120)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1878)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1754)
	at com.sun.enterprise.admin.cli.embeddable.CommandExecutorImpl.executeCommand(CommandExecutorImpl.java:183)
	at com.sun.enterprise.admin.cli.embeddable.CommandExecutorImpl.run(CommandExecutorImpl.java:96)
	at fish.payara.micro.impl.PayaraMicroImpl.bootStrap(PayaraMicroImpl.java:1068)
	at fish.payara.micro.impl.PayaraMicroImpl.create(PayaraMicroImpl.java:216)
	at fish.payara.micro.impl.PayaraMicroImpl.main(PayaraMicroImpl.java:203)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at fish.payara.micro.boot.loader.MainMethodRunner.run(MainMethodRunner.java:50)
	at fish.payara.micro.boot.loader.Launcher.launch(Launcher.java:114)
	at fish.payara.micro.boot.loader.Launcher.launch(Launcher.java:73)
	at fish.payara.micro.boot.PayaraMicroLauncher.create(PayaraMicroLauncher.java:87)
	at fish.payara.micro.boot.PayaraMicroLauncher.main(PayaraMicroLauncher.java:71)
	at fish.payara.micro.PayaraMicro.main(PayaraMicro.java:397)
```

**Comment by rdebusscher** [comment]
Hi,

I get a different error when I deploy the test application to the server (a class cast exception).

So I'm not sure what the problem is at your end, but you can try to add the JVM option `-Djdk.util.jar.enableMultiRelease=force`   to the server config (so that the correct CatalogManager might be used)

You might also need to add the javax.xml.ws classes as they are no longer a part of java SE from Java 11 and onwards.

Regards
Rudy

**Comment by MeroRai** [comment]
Hi @johny65,

Due to the lack of activity form you, I am closing this issue immediately. If the issue is still pressing to you, feel free to verify if it’s applicable in the current release of Payara Server Community edition and proceed to raise a new issue with details of the test reproducer. Many thanks for your understanding.

**Comment by poikilotherm** [comment]
Sorry to be a necro for this old issue.

Today I tried to deploy Arquillian based tests for Dataverse softwareto a Payara Managed instance and I could not deploy my WAR. See relevant Maven output here: https://pastebin.com/sz5im9i8

Using OpenJDK 11, Arquillian 1.7.0.Alpha8, JUnit 5, Payara 5.2021.1, Maven 3.5 here.

After some digging, I found out that our dependency on Apache Tika for fulltext indexing [itself depends on the CXF Rest Client `org.apache.cxf.cxf-rt-rs-client`](https://mvnrepository.com/artifact/org.apache.tika/tika-parsers/1.24.1). Excluding per [this StackOverflow thread](https://stackoverflow.com/questions/34942596/java-ee-7-webapp-problems-with-rest-servicesjson-after-upgrading-tika-parser)
made the tests running.

```
        <dependency>
            <groupId>org.apache.tika</groupId>
            <artifactId>tika-parsers</artifactId>
            <version>1.24.1</version>
            <exclusions>
                <exclusion>
                    <groupId>org.apache.cxf</groupId>
                    <artifactId>cxf-rt-rs-client</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
```

We have `<class-loader delegate="true"/>` in `glassfish.web.xml`, so CXF should not be loaded, shouldn't it?

Now I'm wondering if this is related, if I should post to forum or create a new issue.
(This does not look like an Arquillian problem, as this happened in the past, too.)
